modules/core/src/test/java/org/apache/ignite/internal/processors/cache/checker/processor/PartitionReconciliationLostPartitionsTest.java

```diff
@@ -0,0 +1,140 @@
+import static org.apache.ignite.internal.processors.cache.checker.util.ConsistencyCheckUtils.RECONCILIATION_DIR;
+
+public class PartitionReconciliationLostPartitionsTest extends PartitionReconciliationAbstractTest {
+    /** Nodes. */
+    private static final int NODES_CNT = 2;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setDataStorageConfiguration(new DataStorageConfiguration()
+            .setDefaultDataRegionConfiguration(new DataRegionConfiguration()
+                .setPersistenceEnabled(false)
+                .setMaxSize(300L * 1024 * 1024))
+        );
+
+        CacheConfiguration<Integer, Integer> ccfg = new CacheConfiguration<>();
+
+        ccfg.setName(DEFAULT_CACHE_NAME);
+        ccfg.setAtomicityMode(ATOMIC);
+        ccfg.setWriteSynchronizationMode(CacheWriteSynchronizationMode.FULL_SYNC);
+        ccfg.setAffinity(new RendezvousAffinityFunction(false, 12));
+        ccfg.setBackups(0);
+        ccfg.setPartitionLossPolicy(PartitionLossPolicy.READ_WRITE_SAFE);
+
+        cfg.setCacheConfiguration(ccfg);
+        cfg.setConsistentId(igniteInstanceName);
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        stopAllGrids();
+
+        U.delete(U.resolveWorkDirectory(U.defaultWorkDirectory(), RECONCILIATION_DIR, false));
+    }
+
+    /**
+     * Tests partition reconciliation in the case when a verifying cache has lost partitions.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testLostPartitions() throws Exception {
+        // Start cluster and fill the cache.
+        startGridsMultiThreaded(NODES_CNT);
+
+        grid(0).cluster().state(ACTIVE);
+
+        // Stop node to get lost partitions.
+        stopGrid(1);
+
+        Set<Integer> lostParts = grid(0).cachex(DEFAULT_CACHE_NAME).context().topology().lostPartitions();
+
+        assertFalse("There are no lost partitions!", lostParts.isEmpty());
+
+        ReconciliationResult res = partitionReconciliation(
+            grid(0),
+            new VisorPartitionReconciliationTaskArg.Builder()
+                .locOutput(true)
+                .repair(false)
+                .caches(Collections.singleton(DEFAULT_CACHE_NAME)));
+
+        ReconciliationResult compactRes = partitionReconciliation(
+            grid(0),
+            new VisorPartitionReconciliationTaskArg.Builder()
+                .locOutput(false)
+                .repair(false)
+                .caches(Collections.singleton(DEFAULT_CACHE_NAME)));
+
+        assertEquals(
+            "Number of lost partitions does not match to the number of skipped entities.",
+            res.partitionReconciliationResult().skippedEntriesCount(),
+            lostParts.size());
```

Comment:
Well, I didn't check startGridsMultiThreaded logic. The amount of an expected lost partitions differs from the amount of the caches that have such lost partitions, is that correct?

Reply:
In general, `res.partitionReconciliationResult().skippedEntriesCount()` should return a total number of skipped entries that is the total number of lost partitions for all caches included in the check.